tools/clang/test/HLSLFileCheck/hlsl/template/DependentTypeInObjectParamInvalid.hlsl

```diff
@@ -0,0 +1,21 @@
+// RUN: %dxc -E main -T vs_6_6 -HV 2021 -enable-16bit-types -ast-dump %s | FileCheck %s
+
+// CHECK: error: 'MyStruct' cannot be used as a type parameter where a scalar is required
```

Comment:
I agree, but I could not supply the shader model for the verifier test at the time, so this test wouldn't compile because of the `ResourceDescriptorHeap` usage.

On that note, I had found what looked like a verifier test that it looked like this would fit well into, but it also used `ResourceDescriptorHeap`, and wasn't being used at all (`template-diag-deferred.hlsl`).  I tried adding that to VerifierTest.cpp, along with a RUN line and ran into this trouble.  I also noticed the VerifierHelper.py was out of date for the list of tests, and especially the set of tests that wouldn't be compatible with fxc.

Since then I found another way to formulate the test that doesn't depend on `ResourceDescriptorHeap`, so I'll update it with that.

As a side note, I think we likely need to keep these dynamic resource built-in global identifiers available regardless of shader model, and check shader model to validate legal usage instead.

Reply:
I changed the test and moved it to a verifier test.